### ISSUE TITLE
Uploading package to bintray without recipe is possible

### ISSUE THREAD

**Issue by danimtb** [conan-io/conan#2663]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant: conan 1.1.1
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

*./conanfile.py*
```
from conans import ConanFile

class bugConan(ConanFile):
    name = "bug"
    version = "0.0.0"
```
Conan commands:
```
$ conan create . danimtb/testing
....
...
bug/0.0.0@danimtb/testing: Calling package()
bug/0.0.0@danimtb/testing: WARN: This conanfile has no package step
bug/0.0.0@danimtb/testing: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' created

$ conan upload bug/0.0.0@danimtb/testing -p 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 -r bintray_repo
Requesting upload permissions...Done!
Uploading conanmanifest.txt
[==================================================] 59B/59B
Uploading conaninfo.txt
[==================================================] 150B/150B
Uploading conan_package.tgz
[==================================================] 63B/63B
```
Package uploaded to Bintray without recipe: https://bintray.com/danimtb/public-conan/bug%3Adanimtb#files/danimtb%2Fbug%2F0.0.0%2Ftesting%2Fpackage

If I remove package from local cache and ``conan install`` it again it fails:
```
$ conan remove bug/0.0.0@danimtb/testing
...
$ conan install  bug/0.0.0@danimtb/testing -r bintray_repo
bug/0.0.0@danimtb/testing: Not found, retrieving from server 'bintray_repo'
bug/0.0.0@danimtb/testing: Trying with 'upload_repo'...
ERROR: bug/0.0.0@danimtb/testing was not found in remote 'bintray_repo'
```
Making a search by reference in that same remote also fails:
```
$ conan search bug/0.0.0@danimtb/testing -r upload_repo
ERROR: . [Remote: bintray_repo]
```
But, if I only delete the uploaded package from cache and keep the recipe exported, it works:
```
$ conan remove bug/0.0.0@danimtb/testing -p 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
...
$ conan install  bug/0.0.0@danimtb/testing -r upload_repo
bug/0.0.0@danimtb/testing: Installing package
Requirements
    bug/0.0.0@danimtb/testing from local cache
Packages
    bug/0.0.0@danimtb/testing:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9

bug/0.0.0@danimtb/testing: Retrieving package 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
bug/0.0.0@danimtb/testing: Looking for package 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 in remote 'bintray_repo'
Downloading conanmanifest.txt
[==================================================] 59B/59B
Downloading conaninfo.txt
[==================================================] 150B/150B
Downloading conan_package.tgz
[==================================================] 63B/63B
bug/0.0.0@danimtb/testing: Package installed 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
```

This behavior does not happen with conan server, as it does not allow to upload a pacakge without having a recipe already uploaded.

```
$ conan upload bug/0.0.0@danimtb/testing -p 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 -r local_server
Uploading package 1/1: 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
Requesting upload permissions...                                      ERROR: There are no remote conanfiles like bug/0.0.0@danimtb/testing. [Remote: local_server]
```

**Comment by danimtb** [comment]
A question related to this: Should ``conan upload -p 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 -r remote`` always upload recipe as well as the package?

**Comment by danimtb** [comment]
This issue also affects Artifactory

**Comment by memsharded** [comment]
This should have been fixed in conan 1.5.
